backend/api/Services/DocumentService.cs

```diff
@@ -39,6 +49,45 @@ public IList<PimsActivityInstanceDocument> GetActivityDocuments(long activityId)
             return documentActivityRespostory.GetAll(activityId);
         }
 
+        public async Task<bool> DeleteActivityDocumentAsync(PimsActivityInstanceDocument activityDocument)
+        {
+            IList<PimsActivityInstanceDocument> existingActivityDocuments = documentActivityRespostory.GetAll(activityDocument.ActivityInstanceId);
```

Comment:
Hmm, I don't think this is correct. I thought that the logic for deleting is:
1) If this document is used by **any other activity** then don't delete it, just remove the reference between this activity and the document.
2) otherwise remove the relationship and also delete the document in mayan.

Whereas this logic appears to be saying: if this is the only document associated with this activity, then delete the document and remove the reference. Otherwise only remove the reference